src/store/queue/sagas.ts

```diff
@@ -206,36 +258,22 @@ export function* watchNext() {
       return
     }
 
-    // Skip deleted track
     const id = yield select(getQueueTrackId)
     const track = yield select(getTrack, { id })
+    // Skip deleted track
     if (track && track.is_delete) {
       yield put(next({ skip }))
     } else {
-      const index = yield select(getIndex)
-      if (index >= 0) {
-        const source = yield select(getSource)
-        const length = yield select(getLength)
-        const shuffle = yield select(getShuffle)
-        const repeatMode = yield select(getRepeat)
-
-        const isCloseToEndOfQueue = index + 1 >= length
-        const isNotRepeating =
-          repeatMode === RepeatMode.OFF ||
-          (repeatMode === RepeatMode.SINGLE && skip)
-
-        // Get recommended tracks if not in shuffle and repeat modes and close to end of queue
-        if (!shuffle && isNotRepeating && isCloseToEndOfQueue) {
-          const userId = yield select(getUserId)
-          yield put(
-            queueActions.queueAutoplay(
-              track?.genre,
-              track ? [track.track_id] : [],
-              userId
-            )
-          )
-        }
-        const uid = yield select(getUid)
+      const uid = yield select(getUid)
+      const source = yield select(getSource)
+
+      yield call(handleQueueAutoplay, {
+        skip: !!skip,
+        ignoreSkip: false,
+        track
+      })
+
+      if (track) {
```

Comment:
why the if track check here? is that effectively what we did with the index check before?

Reply:
> The change seems reasonable, but for testing, let's make sure all the following work
> 
> 1. Autoplay works at end of lineup
> 2. Repeat single works at end of song that is first, middle, and end of lineup
> 3. Repeat full works at the end of line up
> 4. Pressing next manually when repeat single is on goes to the next track
> 5. Shuffle works within a lineup

All the above work. Only weird thing I noticed is the playbar line doesn't reset on repeat single, which is an existing problem on master also. looking into it